scripts/benchmark_cortex.py

```diff
@@ -1,235 +1,271 @@
+        count = await acount(prox.eval('inet:ipv4#even'))
+        assert count == self.workfactor // 2
+        return count
+
+    @isatrial
+    async def do05PivotAbsent(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.eval('inet:ipv4#odd -> inet:dns:a'))
+        assert count == 0
+        return self.workfactor // 2
+
+    @isatrial
+    async def do06PivotPresent(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.eval('inet:ipv4#even -> inet:dns:a'))
+        assert count == self.workfactor // 2 + self.workfactor // 10
+        return count
+
+    @isatrial
+    async def do07AddNodes(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.addNodes(self.testdata.asns2))
+        assert count == self.workfactor
+        return count
+
+    @isatrial
+    async def do07AddNodesPresent(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.addNodes(self.testdata.ips))
+        assert count == self.workfactor
+        return count
+
+    @isatrial
+    async def do08LocalAddNodes(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(core.addNodes(self.testdata.asns))
+        assert count == self.workfactor
+        return count
+
+    @isatrial
+    async def do09DelNodes(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.eval('inet:url | delnode'))
+        assert count == 0
+        return self.workfactor
+
+    async def run(self, core: 's_cortex.Cortex', name: str, dirn: str, coro) -> None:
+        for i in range(self.num_iters):
+            # We set up the cortex each time to avoid intra-cortex caching
+            # (there's still a substantial amount of OS caching)
+            async with self.getCortexAndProxy(dirn) as (core, prox):
+                gc.collect()
+
+                start = time.time()
+                count = await coro(core, prox)
+                self.measurements[name].append((time.time() - start, count))
+                # yappi.get_func_stats().print_all()
+
+    def _getTrialFuncs(self):
+        funcs: List[Tuple[str, Callable]] = []
+        funcnames = sorted(f for f in dir(self))
+        for funcname in funcnames:
+            func = getattr(self, funcname)
+            if not hasattr(func, '_isatrial'):
+                continue
+            funcs.append((funcname, func))
+        return funcs
+
+    async def runSuite(self, config: Dict[str, Dict], numprocs: int, tmpdir: str = None):
+        assert numprocs == 1
+        if tmpdir is not None:
+            tmpdir = os.path.abspath(tmpdir)
+        with syntest.getTestDir(tmpdir) as dirn:
+            async with self.getCortexAndProxy(dirn) as (core, _):
+                await self._loadtestdata(core)
+            for funcname, func in self._getTrialFuncs():
```

Comment:
Would be useful to print to the user that we're 1. loading test data 2. starting to execute trial funcs.